### PR TITLE
[TASK] Convert array to bracket notation

### PR DESCRIPTION
**Pull request by thomashohn**
Should solve the rest of #50

### CODE REVIEW

File: Classes/Service/Tika/ServerService.php [hunk 1]
```diff
@@ -260,21 +260,23 @@ protected function queryTika($endpoint, $context = null)
...
-        $context = stream_context_create(array(
-            'http' => array(
-                'protocol_version' => 1.1,
-                'method' => 'PUT',
-                'header' => implode(CRLF, $headers),
-                'content' => $file->getContents()
-            )
-        ));
+        $context = stream_context_create(
+            [
```
Comment (irnnr) [inline Classes/Service/Tika/ServerService.php]: I'd keep those on the previous line

File: Classes/Service/Tika/ServerService.php [hunk 2]
```diff
@@ -289,21 +291,23 @@ public function extractText(FileInterface $file)
...
```
Comment (irnnr) [inline Classes/Service/Tika/ServerService.php]: prev line

File: Classes/Service/Tika/ServerService.php [hunk 3]
```diff
@@ -319,20 +323,22 @@ public function extractMetaData(FileInterface $file)
...
```
Comment (irnnr) [inline Classes/Service/Tika/ServerService.php]: prev line

File: Classes/Service/Tika/ServerService.php [hunk 4]
```diff
@@ -347,20 +353,22 @@ public function detectLanguageFromFile(FileInterface $file)
...
-        $context = stream_context_create(array(
-            'http' => array(
-                'protocol_version' => 1.1,
-                'method' => 'PUT',
-                'header' => implode(CRLF, $headers),
-                'content' => $input
-            )
-        ));
+        $context = stream_context_create(
+            [
```
Comment (irnnr) [inline Classes/Service/Tika/ServerService.php]: prev line

File: Classes/Service/Tika/ServerService.php [hunk 5]
```diff
@@ -386,20 +394,21 @@ public function getSupportedMimeTypes()
...
             'Content-Type: application/octet-stream',
             'Accept: application/json',
             'Connection: close'
-        );
+        ];
 
-        $context = stream_context_create(array(
-                'http' => array(
+        $context = stream_context_create(
+            [
```
Comment (irnnr) [inline Classes/Service/Tika/ServerService.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 6]
```diff
@@ -80,18 +80,20 @@ public function setUp()
...
 
         // Disable xml2array cache used by ResourceFactory
-        GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Cache\\CacheManager')->setCacheConfigurations(array(
-            'cache_hash' => array(
-                'frontend' => 'TYPO3\\CMS\\Core\\Cache\\Frontend\\VariableFrontend',
-                'backend' => 'TYPO3\\CMS\\Core\\Cache\\Backend\\TransientMemoryBackend'
-            )
-        ));
+        GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Cache\\CacheManager')->setCacheConfigurations(
+            [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 7]
```diff
@@ -101,26 +103,30 @@ protected function setUpDocumentsStorageMock()
         $this->testDocumentsPath = ExtensionManagementUtility::extPath('tika')
             . 'Tests/TestDocuments/';
 
-        $documentsDriver = $this->createDriverFixture(array(
-            'basePath' => $this->testDocumentsPath,
-            'caseSensitive' => true
-        ));
+        $documentsDriver = $this->createDriverFixture(
+            [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 8]
```diff
@@ -101,26 +103,30 @@ protected function setUpDocumentsStorageMock()
...
             'is_browsable' => true,
             'is_online' => true,
-            'configuration' => $this->convertConfigurationArrayToFlexformXml(array(
-                'basePath' => $this->testDocumentsPath,
-                'pathType' => 'absolute',
-                'caseSensitive' => '1'
-            ))
-        );
+            'configuration' => $this->convertConfigurationArrayToFlexformXml(
+                [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 9]
```diff
@@ -129,26 +135,30 @@ protected function setUpLanguagesStorageMock()
         $this->testLanguagesPath = ExtensionManagementUtility::extPath('tika')
             . 'Tests/TestLanguages/';
 
-        $languagesDriver = $this->createDriverFixture(array(
-            'basePath' => $this->testLanguagesPath,
-            'caseSensitive' => true
-        ));
+        $languagesDriver = $this->createDriverFixture(
+            [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 10]
```diff
@@ -129,26 +135,30 @@ protected function setUpLanguagesStorageMock()
...
             'is_browsable' => true,
             'is_online' => true,
-            'configuration' => $this->convertConfigurationArrayToFlexformXml(array(
-                'basePath' => $this->testLanguagesPath,
-                'pathType' => 'absolute',
-                'caseSensitive' => '1'
-            ))
-        );
+            'configuration' => $this->convertConfigurationArrayToFlexformXml(
+                [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 11]
```diff
@@ -195,9 +205,17 @@ protected function createDriverFixture(
     protected function convertConfigurationArrayToFlexformXml(
         array $configuration
     ) {
-        $flexformArray = array('data' => array('sDEF' => array('lDEF' => [])));
+        $flexformArray = [
+            'data' =>
+                [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 12]
```diff
@@ -195,9 +205,17 @@ protected function createDriverFixture(
...
+                    'sDEF' =>
+                        [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 13]
```diff
@@ -227,10 +245,10 @@ public function extractsMetaDataFromDocFile()
         $service = new ServerService($this->getTikaServerConfiguration());
 
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 14]
```diff
@@ -259,10 +277,10 @@ public function extractsTextFromDocFile()
...
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 15]
```diff
@@ -280,10 +298,10 @@ public function extractsTextFromZipFile()
...
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 16]
```diff
@@ -330,10 +348,10 @@ public function detectsLanguageFromFile($language)
...
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Unit/Backend/SolrModule/TikaControlPanelModuleControllerTest.php [hunk 17]
```diff
@@ -63,17 +63,20 @@ public function canShowInformationFromStandaloneTikaServer()
 
         $this->controller->setTikaService($tikaServerService);
         $this->controller->setTikaConfiguration(
-            array('extractor' => 'server', 'tikaServerPath' => $this->getFixturePath('fake-server-jar.jar'))
+            [
```
Comment (irnnr) [inline Tests/Unit/Backend/SolrModule/TikaControlPanelModuleControllerTest.php]: prev line

File: Tests/Unit/Backend/SolrModule/TikaControlPanelModuleControllerTest.php [hunk 18]
```diff
@@ -63,17 +63,20 @@ public function canShowInformationFromStandaloneTikaServer()
...
+                'extractor' => 'server',
+                'tikaServerPath' => $this->getFixturePath('fake-server-jar.jar')
+            ]
         );
 
         $this->viewMock->expects($this->at(2))->method('assign')->with('server',
```
Comment (irnnr) [inline Tests/Unit/Backend/SolrModule/TikaControlPanelModuleControllerTest.php]: Here, I'd actually move `'server'` to the next line, have it align with the opening square bracket of the array

File: Tests/Unit/Service/Tika/AppServiceTest.php [hunk 19]
```diff
@@ -59,10 +59,10 @@ public function getTikaVersionUsesVParameter()
     public function extractTextUsesTParameter()
     {
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/AppServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/AppServiceTest.php [hunk 20]
```diff
@@ -77,12 +77,12 @@ public function extractTextUsesTParameter()
      */
     public function extractMetaDataUsesMParameter()
     {
-        ExecRecorder::setReturnExecOutput(array('foo'));
+        ExecRecorder::setReturnExecOutput(['foo']);
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/AppServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/AppServiceTest.php [hunk 21]
```diff
@@ -98,10 +98,10 @@ public function extractMetaDataUsesMParameter()
     public function detectLanguageFromFileUsesLParameter()
     {
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/AppServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/ServerServiceTest.php [hunk 22]
```diff
@@ -231,10 +231,10 @@ public function extractTextQueriesTikaEndpoint()
     public function extractMetaDataQueriesMetaEndpoint()
     {
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/ServerServiceTest.php [hunk 23]
```diff
@@ -250,10 +250,10 @@ public function extractMetaDataQueriesMetaEndpoint()
     public function detectLanguageFromFileQueriesLanguageStreamEndpoint()
     {
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServerServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/ServiceUnitTestCase.php [hunk 24]
```diff
@@ -99,26 +101,30 @@ protected function setUpDocumentsStorageMock()
         $this->testDocumentsPath = ExtensionManagementUtility::extPath('tika')
             . 'Tests/TestDocuments/';
 
-        $documentsDriver = $this->createDriverFixture(array(
-            'basePath' => $this->testDocumentsPath,
-            'caseSensitive' => true
-        ));
+        $documentsDriver = $this->createDriverFixture(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServiceUnitTestCase.php]: prev line

File: Tests/Unit/Service/Tika/ServiceUnitTestCase.php [hunk 25]
```diff
@@ -99,26 +101,30 @@ protected function setUpDocumentsStorageMock()
...
             'is_browsable' => true,
             'is_online' => true,
-            'configuration' => $this->convertConfigurationArrayToFlexformXml(array(
-                'basePath' => $this->testDocumentsPath,
-                'pathType' => 'absolute',
-                'caseSensitive' => '1'
-            ))
-        );
+            'configuration' => $this->convertConfigurationArrayToFlexformXml(
+                [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServiceUnitTestCase.php]: prev line

File: Tests/Unit/Service/Tika/ServiceUnitTestCase.php [hunk 26]
```diff
@@ -127,26 +133,30 @@ protected function setUpLanguagesStorageMock()
         $this->testLanguagesPath = ExtensionManagementUtility::extPath('tika')
             . 'Tests/TestLanguages/';
 
-        $languagesDriver = $this->createDriverFixture(array(
-            'basePath' => $this->testLanguagesPath,
-            'caseSensitive' => true
-        ));
+        $languagesDriver = $this->createDriverFixture(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServiceUnitTestCase.php]: prev line

File: Tests/Unit/Service/Tika/ServiceUnitTestCase.php [hunk 27]
```diff
@@ -127,26 +133,30 @@ protected function setUpLanguagesStorageMock()
...
             'is_browsable' => true,
             'is_online' => true,
-            'configuration' => $this->convertConfigurationArrayToFlexformXml(array(
-                'basePath' => $this->testLanguagesPath,
-                'pathType' => 'absolute',
-                'caseSensitive' => '1'
-            ))
-        );
+            'configuration' => $this->convertConfigurationArrayToFlexformXml(
+                [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServiceUnitTestCase.php]: prev line

File: Tests/Unit/Service/Tika/ServiceUnitTestCase.php [hunk 28]
```diff
@@ -193,9 +203,17 @@ protected function createDriverFixture(
     protected function convertConfigurationArrayToFlexformXml(
         array $configuration
     ) {
-        $flexformArray = array('data' => array('sDEF' => array('lDEF' => [])));
+        $flexformArray = [
+            'data' =>
+                [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServiceUnitTestCase.php]: prev line

File: Tests/Unit/Service/Tika/ServiceUnitTestCase.php [hunk 29]
```diff
@@ -193,9 +203,17 @@ protected function createDriverFixture(
...
+                    'sDEF' =>
+                        [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/ServiceUnitTestCase.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 30]
```diff
@@ -80,19 +80,22 @@ public function extractByQueryTextReturnsTextElementFromResponse()
     {
         $expectedValue = 'extracted text element';
         $solrMock = $this->prophet->prophesize('ApacheSolrForTypo3\\Solr\\SolrService');
-        $solrMock->extractByQuery(Argument::type('ApacheSolrForTypo3\\Tika\\Service\\Tika\\SolrCellQuery'))->willReturn(array(
-            $expectedValue,     // extracted text is index 0
-            'meta data element' // meta data is index 1
-        ));
+        $solrMock->extractByQuery(Argument::type('ApacheSolrForTypo3\\Tika\\Service\\Tika\\SolrCellQuery'))
+            ->willReturn(
+                [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 31]
```diff
@@ -80,19 +80,22 @@ public function extractByQueryTextReturnsTextElementFromResponse()
...
+                    'meta data element' // meta data is index 1
+                ]
+            );
 
         $service = new SolrCellService($this->getConfiguration());
         $this->inject($service, 'solr', $solrMock->reveal());
 
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 32]
```diff
@@ -113,10 +116,10 @@ public function extractByQueryTextUsesSolrCellQuery()
...
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 33]
```diff
@@ -129,16 +132,16 @@ public function extractByQueryTextUsesSolrCellQuery()
...
-            ->setConstructorArgs(array($this->getConfiguration()))
-            ->setMethods(array('cleanupTempFile'))
+            ->setConstructorArgs([$this->getConfiguration()])
+            ->setMethods(['cleanupTempFile'])
             ->getMock();
         $serviceMock->expects($this->once())->method('cleanupTempFile');
 
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 34]
```diff
@@ -155,19 +158,21 @@ public function extractMetaDataUsesSolrCellQuery()
         $solrMock = $this->prophet->prophesize('ApacheSolrForTypo3\\Solr\\SolrService');
         $solrMock->extractByQuery(Argument::type('ApacheSolrForTypo3\\Tika\\Service\\Tika\\SolrCellQuery'))
             ->shouldBeCalled()
-            ->willReturn(array(
-                'foo',       // extracted text is index 0
-                array('bar') // meta data is index 1
-            ));
+            ->willReturn(
+                [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 35]
```diff
@@ -155,19 +158,21 @@ public function extractMetaDataUsesSolrCellQuery()
...
+                    'foo', // extracted text is index 0
+                    ['bar'] // meta data is index 1
+                ]
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: next line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 36]
```diff
@@ -155,19 +158,21 @@ public function extractMetaDataUsesSolrCellQuery()
...
+            );
 
         $service = new SolrCellService($this->getConfiguration());
         $this->inject($service, 'solr', $solrMock->reveal());
 
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: Tests/Unit/Service/Tika/SolrCellServiceTest.php [hunk 37]
```diff
@@ -180,16 +185,16 @@ public function extractMetaDataUsesSolrCellQuery()
...
-            ->setConstructorArgs(array($this->getConfiguration()))
-            ->setMethods(array('cleanupTempFile'))
+            ->setConstructorArgs([$this->getConfiguration()])
+            ->setMethods(['cleanupTempFile'])
             ->getMock();
         $serviceMock->expects($this->once())->method('cleanupTempFile');
 
         $file = new File(
-            array(
+            [
```
Comment (irnnr) [inline Tests/Unit/Service/Tika/SolrCellServiceTest.php]: prev line

File: ext_tables.php [hunk 38]
```diff
@@ -18,7 +18,7 @@
     $iconRegistry = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\CMS\Core\Imaging\IconRegistry');
 
     $iconRegistry->registerIcon($modulePrefix . '-tikacontrolpanel', $bitmapProvider,
```
Comment (irnnr) [inline ext_tables.php]: when breaking arguments onto a new line, they should **all** be one per line

**Comment by thomashohn** [comment]
@irnnr could you have a look at this one again :)

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 39]
```diff
@@ -101,26 +102,28 @@ protected function setUpDocumentsStorageMock()
         $this->testDocumentsPath = ExtensionManagementUtility::extPath('tika')
             . 'Tests/TestDocuments/';
 
-        $documentsDriver = $this->createDriverFixture(array(
-            'basePath' => $this->testDocumentsPath,
-            'caseSensitive' => true
-        ));
+        $documentsDriver = $this->createDriverFixture([
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: You open both the bracket and the square bracket on the same line.

File: Tests/Integration/Service/Tika/ServerServiceTest.php [hunk 40]
```diff
@@ -101,26 +102,28 @@ protected function setUpDocumentsStorageMock()
...
+                'basePath' => $this->testDocumentsPath,
+                'caseSensitive' => true
+            ]
+        );
```
Comment (irnnr) [inline Tests/Integration/Service/Tika/ServerServiceTest.php]: thus this closing bracket should be on the previous line together with the closing square bracket, keeping the same indentation levels.

There were a couple more of these. I went ahead and fixed them.

File: Tests/Unit/Service/Tika/AppServiceTest.php (hunk 19 above)
Comment (irnnr) [inline Tests/Unit/Service/Tika/AppServiceTest.php]: I was wrong on these, sorry about that!

**Comment by irnnr** [comment]
I was wrong on these, sorry about that! I fixed them.

**Pull request merged**